### PR TITLE
Harmonize prior and posterior predictive plots in MMM module

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/pymc-labs/pymc-marketing/issues/1144

I simply extend the posterior predictive functionality to the prior predictive case without copy-paste but rather abstracting the plotting function by "group".

----
📚 Documentation preview 📚: https://pymc-marketing--1170.org.readthedocs.build/en/1170/

### CODE REVIEW

**Comment by juanitorduz** [comment]
@wd60622 Is https://github.com/pymc-labs/pymc-marketing/actions/runs/11685938935/job/32540558097?pr=1170 related to a clv notebook error 🤔 ?

**Comment by juanitorduz** [comment]
> Can the end-to-end MM example also be updated to use this when conducting the prior predictive check?

Addded in https://github.com/pymc-labs/pymc-marketing/pull/1170/commits/53acf0845c0d144b9e2c91b849bc75b0cd99c441

**Comment by cluhmann** [comment]
Can I ask why there is a mean plotted in the `sample_prior_predictive()` plot and why it's approximately zero (this is in the end to end MMM notebook)?

https://pymc-marketing--1170.org.readthedocs.build/en/1170/_images/2f0680c00d41aaca16ccd466cb2d48ba317aeef9f046c9da07ecaa81bfe225aa.png

**Comment by cluhmann** [comment]
> (this is in the end to end MMM notebook)?

There is also an "observed" series in the MMM example notebook prior predictive plot.

**Comment by juanitorduz** [comment]
> Can I ask why there is an "observed" series plotted in the `sample_prior_predictive()` plot and why it's approximately zero (this is in the end to end MMM notebook)?
> 
> https://pymc-marketing--1170.org.readthedocs.build/en/1170/_images/2f0680c00d41aaca16ccd466cb2d48ba317aeef9f046c9da07ecaa81bfe225aa.png

@cluhmann this is the observed series we (I?) usually compare with when I am doing the prior-predictive checks.

**Comment by cluhmann** [comment]
Ah. So the observed series looks like it's zero because it's variability is an order of magnitude smaller than the prior predictive plot?

**Comment by juanitorduz** [comment]
Exactly! I'll probably make the prior stricter.

**Comment by cluhmann** [comment]
I should have realized that. I need more coffee.

**Comment by juanitorduz** [comment]
@cluhmann can we merge this one so that I can tackle https://github.com/pymc-labs/pymc-marketing/issues/1176 (git with. notebooks is a pain ;) )

**Comment by cluhmann** [comment]
I am rushing to get my pymc-marketing talk done (which is why I am running into these little documentation issues), so I can't give this a thorough review right now. But I probably could tomorrow (or someone like @wd60622 could). Sorry :grimacing:

**Comment by juanitorduz** [comment]
all good! I will start solving in another branch the other issues (good luck in the talk ;)  and thanks for all the feedback)

**Comment by wd60622** [comment]
I will give review today when I can

File: docs/source/notebooks/mmm/mmm_case_study.ipynb [hunk 1]
```diff
@@ -45,7 +45,7 @@
   },
```
Comment (wd60622) [inline docs/source/notebooks/mmm/mmm_case_study.ipynb:45]: <p>Can we change the verbiage here as well. The use of "reasonable" is a bit strange </p><p> </p><p>Also, would we want to comment on the fact that the prior predictive includes negative values... That might element to highlight</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

File: tests/mmm/test_base.py [hunk 2]
```diff
@@ -342,17 +342,25 @@ def test_calling_prior_before_sample_prior_predictive_raises_error(
...
-        match="Make sure the model has bin fitted and the posterior predictive has been sampled!",
+        match="Make sure the model has been fitted and the posterior_predictive has been sampled!",
     ):
         test_mmm.plot_posterior_predictive()
 
 
 def test_get_errors_raises_not_fitted(test_mmm) -> None:
     with pytest.raises(
         RuntimeError,
-        match="Make sure the model has bin fitted and the posterior predictive has been sampled!",
```
Comment (wd60622) [inline tests/mmm/test_base.py:356]: testing for the typo 😆

**Review by wd60622 — COMMENTED** [review]
Just some commentary on the verbiage around the stuff mainly. Do you think the Likelihood distribution is something that we would like to change? Maybe first here in the notebook then in the defaults

WDYT?

**Review by wd60622 — APPROVED** [review]

**Comment by juanitorduz** [comment]
> Just some commentary on the verbiage around the stuff mainly. Do you think the Likelihood distribution is something that we would like to change? Maybe first here in the notebook then in the defaults
> 
> WDYT?

Yes, let's look this into next iterations.

**Comment by juanitorduz** [comment]
@wd60622 I addressed your comments :)

**Comment by wd60622** [comment]
> @wd60622 I addressed your comments :)

Let's ship it!

**Comment by wd60622** [comment]
Failing because of this: https://github.com/pymc-labs/pymc-marketing/issues/1148

**Pull request merged**